홈 대시보드 배너·시장온도·관심종목·시그널보드 렌더링/판단 오류

## 배경
홈 대시보드 첫 화면에서 사용자가 바로 보는 핵심 위젯들에 몇 가지 품질 이슈가 있습니다.

## 문제
- 상단 검정색 롤링 배너가 일부 환경에서 정상적으로 흘러가지 않고 레이아웃이 불안정함
- 시장온도가 실제 체감보다 과하게 강세/강한 강세로 표시되는 경우가 잦음
- 시장온도가 시그널 생성 타이밍에 과도하게 의존해 늦게 갱신되는 경우가 있음
- 관심종목 이름이 모바일에서 강제로 잘려 노출됨
- 시그널보드에서 동일 코인(예: 이더리움)이 중복 노출되거나 코인 심볼 이미지가 불안정하게 보임

## 원인
- 가 롤링 트랙 자체를 뷰포트처럼 사용하고 있어 overflow/width 제어가 불안정함
- 시장온도 로직이 시그널 강도 합계에 크게 의존하고, 최신성/표본 수 보정이 약함
- 가격 기반 fallback이 전체 자산 단순 평균이라 일부 자산군의 과열이 전체 온도를 끌어올릴 수 있음
- 관심종목 모바일 칩이 이름을 고정 길이로 잘라 렌더링함
- 가 아이템 변경 시 실패 상태를 리셋하지 않아 재사용 렌더에서 로고 fallback 상태가 남을 수 있음
- 시그널보드가 동일 심볼의 다중 시그널을 그대로 보여 요약 보드 역할과 맞지 않음

## 기대 동작
- 상단 롤링 배너는 라벨 영역과 트랙 뷰포트가 분리되어 안정적으로 스크롤되어야 함
- 시장온도는 시그널 최신성/표본 수/시장별 가격 흐름을 함께 반영해 보수적으로 표시되어야 함
- 관심종목 이름은 잘리지 않더라도 가독성 있게 노출되어야 함
- 시그널보드는 동일 심볼 기준 대표 시그널 중심으로 보여야 하고, 주요 코인 로고 fallback이 안정적이어야 함

## QA
- 
> market-dashboard-v2@0.0.0 build
> vite build

vite v8.0.0 building client environment for production...
[2K
transforming...✓ 2234 modules transformed.
rendering chunks...
computing gzip size...
dist/registerSW.js                          0.13 kB
dist/manifest.webmanifest                   0.91 kB
dist/index.html                             1.78 kB │ gzip:   0.86 kB
dist/assets/index-lGtevIr0.css             39.74 kB │ gzip:   8.29 kB
dist/assets/rolldown-runtime-DF2fYuay.js    0.55 kB │ gzip:   0.35 kB
dist/assets/query-DJUBvpWF.js              29.82 kB │ gzip:   9.97 kB
dist/assets/charts-DzZessfj.js            163.71 kB │ gzip:  52.54 kB
dist/assets/react-kGcS_FM1.js             184.19 kB │ gzip:  58.42 kB
dist/assets/index-D81RvyDy.js             607.24 kB │ gzip: 185.07 kB

✓ built in 539ms

PWA v1.2.0
mode      generateSW
precache  15 entries (1013.55 KiB)
files generated
  dist/sw.js
  dist/workbox-61fbeec5.js 통과
- 
> market-dashboard-v2@0.0.0 test
> vitest run


 RUN  v4.1.1 /Users/bong/market-dashboard-v5


 Test Files  2 passed (2)
      Tests  39 passed (39)
   Start at  02:48:14
   Duration  114ms (transform 32ms, setup 0ms, import 44ms, tests 4ms, environment 0ms) 통과 (39 passed)
- 
> market-dashboard-v2@0.0.0 lint
> eslint .


/Users/bong/market-dashboard-v5/api/_price-cache.js
  116:18  error  '_' is defined but never used  no-unused-vars

/Users/bong/market-dashboard-v5/api/cron/update-coins.js
  262:83  error  '_' is defined but never used  no-unused-vars

/Users/bong/market-dashboard-v5/api/cron/update-kr.js
    6:40  error  'getSnap' is defined but never used. Allowed unused vars must match /^[A-Z_]/u  no-unused-vars
  427:97  error  '_' is defined but never used                                                   no-unused-vars

/Users/bong/market-dashboard-v5/api/cron/update-us.js
   90:14  error  '_' is defined but never used  no-unused-vars
  112:16  error  '_' is defined but never used  no-unused-vars
  235:16  error  '_' is defined but never used  no-unused-vars
  271:16  error  '_' is defined but never used  no-unused-vars
  281:16  error  '_' is defined but never used  no-unused-vars
  286:74  error  '_' is defined but never used  no-unused-vars
  302:80  error  '_' is defined but never used  no-unused-vars

/Users/bong/market-dashboard-v5/api/cron/whale-telegram.js
  134:92  error  '_' is defined but never used  no-unused-vars

/Users/bong/market-dashboard-v5/api/morning-briefing.js
  100:39  error  'req' is defined but never used. Allowed unused args must match /^_/u  no-unused-vars
  148:12  error  'err' is defined but never used                                        no-unused-vars

/Users/bong/market-dashboard-v5/api/pcr.js
  4:39  error  'request' is defined but never used. Allowed unused args must match /^_/u  no-unused-vars

/Users/bong/market-dashboard-v5/src/App.jsx
  115:6  warning  React Hook useCallback has a missing dependency: 'setKrStocks'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  126:6  warning  React Hook useCallback has a missing dependency: 'setUsStocks'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bong/market-dashboard-v5/src/components/ChartSidePanel.jsx
  340:6   warning  React Hook useEffect has a missing dependency: 'isIntraday'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       react-hooks/exhaustive-deps
  469:47  warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  467 |
  468 |   useEffect(() => {
> 469 |     if (!symbol || !/^\d{6}$/.test(symbol)) { setLoading(false); return; }
      |                                               ^^^^^^^^^^ Avoid calling setState() directly within an effect
  470 |     setLoading(true);
  471 |     Promise.all([
  472 |       fetchInvestorDataSafe(symbol),  react-hooks/set-state-in-effect
  675:3   warning  Unused eslint-disable directive (no problems were reported from 'no-param-reassign')
  721:6   warning  React Hook useMemo has a missing dependency: 'item'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               react-hooks/exhaustive-deps
  811:6   warning  React Hook useEffect has a missing dependency: 'item'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             react-hooks/exhaustive-deps

/Users/bong/market-dashboard-v5/src/components/GlobalSearch.jsx
  187:6  warning  React Hook useMemo has a missing dependency: 'usSearchItems'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bong/market-dashboard-v5/src/components/Header.jsx
  30:3   error  'krStocks' is assigned a value but never used. Allowed unused args must match /^_/u  no-unused-vars
  30:18  error  'usStocks' is assigned a value but never used. Allowed unused args must match /^_/u  no-unused-vars
  30:33  error  'coins' is assigned a value but never used. Allowed unused args must match /^_/u     no-unused-vars

/Users/bong/market-dashboard-v5/src/components/HomeDashboard.jsx
  487:20  warning  Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

  485 |     return allNews.filter(n => {
  486 |       if (!n.pubDate) return false;
> 487 |       try { return Date.now() - new Date(n.pubDate).getTime() < cutoff; }
      |                    ^^^^^^^^^^ Cannot call impure function
  488 |       catch { return false; }
  489 |     });
  490 |   }, [allNews]);  react-hooks/purity
  559:5   warning  React Hook useMemo has a missing dependency: 'isWatched'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              react-hooks/exhaustive-deps

/Users/bong/market-dashboard-v5/src/components/InvestorFlow.jsx
  41:7  warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  39 |   useEffect(() => {
  40 |     if (!symbol || !/^\d{6}$/.test(symbol)) {
> 41 |       setLoading(false);
     |       ^^^^^^^^^^ Avoid calling setState() directly within an effect
  42 |       return;
  43 |     }
  44 |     setLoading(true);  react-hooks/set-state-in-effect

/Users/bong/market-dashboard-v5/src/components/NewsSidePanel.jsx
  141:7  warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  139 |     // 캐시 히트: 로딩 없이 즉시 표시
  140 |     if (_summaryCache.has(news.link)) {
> 141 |       setAiSummary(_summaryCache.get(news.link));
      |       ^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  142 |       setSummaryLoading(false);
  143 |       return;
  144 |     }  react-hooks/set-state-in-effect
  161:6  warning  React Hook useEffect has missing dependencies: 'news.description', 'news.summary', and 'news.title'. Either include them or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   react-hooks/exhaustive-deps

/Users/bong/market-dashboard-v5/src/components/UnifiedFeedPanel.jsx
  202:45  warning  Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

  200 |     return allSignals.slice(0, 15).map(sig => ({
  201 |       feedType: 'signal',
> 202 |       ts: sig.timestamp || sig.createdAt || Date.now(),
      |                                             ^^^^^^^^^^ Cannot call impure function
  203 |       data: sig,
  204 |       id: `sig-${sig.id}`,
  205 |     })).sort((a, b) => {  react-hooks/purity

/Users/bong/market-dashboard-v5/src/components/WatchlistTable.jsx
  623:9   error    'hot5Count' is assigned a value but never used. Allowed unused vars must match /^[A-Z_]/u                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       no-unused-vars
  655:26  warning  Compilation Skipped: Use of incompatible library

This API returns functions which cannot be memoized without leading to stale UI. To prevent this, by default React Compiler will skip memoizing this component/hook. However, you may see issues if values from this API are passed to other components/hooks that are memoized.

  653 |
  654 |   // 가상 스크롤 — 화면에 보이는 행만 렌더링 (4,000+ 종목 성능 최적화)
> 655 |   const rowVirtualizer = useVirtualizer({
      |                          ^^^^^^^^^^^^^^ TanStack Virtual's `useVirtualizer()` API returns functions that cannot be memoized safely
  656 |     count: isInitialLoad ? 8 : totalCount,
  657 |     getScrollElement: () => scrollContainerRef.current,
  658 |     estimateSize: () => 64, // FlashRow 높이 기준 (px)  react-hooks/incompatible-library

/Users/bong/market-dashboard-v5/src/components/WhalePanel.jsx
  440:5  warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  438 |     if (!isVisible) return;
  439 |
> 440 |     setBtcConnected(false);
      |     ^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  441 |
  442 |     // ── Blockchain.com BTC 온체인 WebSocket ──────────────────────
  443 |     subscribeBtcWhales((evt) => {  react-hooks/set-state-in-effect
  465:6  warning  React Hook useEffect has a missing dependency: 'addEvent'. Either include it or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    react-hooks/exhaustive-deps
  548:6  warning  React Hook useEffect has missing dependencies: 'addEvent' and 'seenChainIds'. Either include them or remove the dependency array                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               react-hooks/exhaustive-deps

/Users/bong/market-dashboard-v5/src/components/home/AiDebateSection.jsx
  48:77  error  'krStocks' is assigned a value but never used. Allowed unused args must match /^_/u  no-unused-vars

/Users/bong/market-dashboard-v5/src/components/home/EventTicker.jsx
  140:21  error  'cfg' is assigned a value but never used. Allowed unused vars must match /^[A-Z_]/u  no-unused-vars

/Users/bong/market-dashboard-v5/src/components/home/HotListSection.jsx
  113:26  warning  Error: Cannot create components during render

Components created during render will reset their state each time they are created. Declare components outside of render.

  111 |           </div>
  112 |           <div className="py-1">
> 113 |             {!krOpen && <ClosedLabel label={krLabel} />}
      |                          ^^^^^^^^^^^ This component is created during render
  114 |             {!hasData
  115 |               ? <SkeletonHotRow count={5} />
  116 |               : krHot.length > 0

  89 |
  90 |   // 휴장 시 종목 리스트 위에 라벨 표시 (종목은 딤 처리로 유지)
> 91 |   const ClosedLabel = ({ label }) => (
     |                       ^^^^^^^^^^^^^^^^
> 92 |     <div className="px-4 pt-2 pb-1 flex items-center gap-1.5">
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 93 |       <span className="text-[10px] text-[#B0B8C1]">🌙 {label}</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 94 |       <span className="text-[10px] text-[#C9CDD2]">· 전일 종가 기준</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 95 |     </div>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 96 |   );
     | ^^^^ The component is created during render here
  97 |
  98 |   return (
  99 |     <>           react-hooks/static-components
  142:26  warning  Error: Cannot create components during render

Components created during render will reset their state each time they are created. Declare components outside of render.

  140 |           </div>
  141 |           <div className="py-1">
> 142 |             {!usOpen && <ClosedLabel label={usLabel} />}
      |                          ^^^^^^^^^^^ This component is created during render
  143 |             {!hasData
  144 |               ? <SkeletonHotRow count={5} />
  145 |               : usHot.length > 0

  89 |
  90 |   // 휴장 시 종목 리스트 위에 라벨 표시 (종목은 딤 처리로 유지)
> 91 |   const ClosedLabel = ({ label }) => (
     |                       ^^^^^^^^^^^^^^^^
> 92 |     <div className="px-4 pt-2 pb-1 flex items-center gap-1.5">
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 93 |       <span className="text-[10px] text-[#B0B8C1]">🌙 {label}</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 94 |       <span className="text-[10px] text-[#C9CDD2]">· 전일 종가 기준</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 95 |     </div>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 96 |   );
     | ^^^^ The component is created during render here
  97 |
  98 |   return (
  99 |     <>           react-hooks/static-components
  201:26  warning  Error: Cannot create components during render

Components created during render will reset their state each time they are created. Declare components outside of render.

  199 |           </div>
  200 |           <div className="py-1">
> 201 |             {!krOpen && <ClosedLabel label={krLabel} />}
      |                          ^^^^^^^^^^^ This component is created during render
  202 |             {!hasData
  203 |               ? <SkeletonHotRow count={5} />
  204 |               : krDrop.map((item, i) => (

  89 |
  90 |   // 휴장 시 종목 리스트 위에 라벨 표시 (종목은 딤 처리로 유지)
> 91 |   const ClosedLabel = ({ label }) => (
     |                       ^^^^^^^^^^^^^^^^
> 92 |     <div className="px-4 pt-2 pb-1 flex items-center gap-1.5">
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 93 |       <span className="text-[10px] text-[#B0B8C1]">🌙 {label}</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 94 |       <span className="text-[10px] text-[#C9CDD2]">· 전일 종가 기준</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 95 |     </div>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 96 |   );
     | ^^^^ The component is created during render here
  97 |
  98 |   return (
  99 |     <>  react-hooks/static-components
  228:26  warning  Error: Cannot create components during render

Components created during render will reset their state each time they are created. Declare components outside of render.

  226 |           </div>
  227 |           <div className="py-1">
> 228 |             {!usOpen && <ClosedLabel label={usLabel} />}
      |                          ^^^^^^^^^^^ This component is created during render
  229 |             {!hasData
  230 |               ? <SkeletonHotRow count={5} />
  231 |               : usDrop.map((item, i) => (

  89 |
  90 |   // 휴장 시 종목 리스트 위에 라벨 표시 (종목은 딤 처리로 유지)
> 91 |   const ClosedLabel = ({ label }) => (
     |                       ^^^^^^^^^^^^^^^^
> 92 |     <div className="px-4 pt-2 pb-1 flex items-center gap-1.5">
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 93 |       <span className="text-[10px] text-[#B0B8C1]">🌙 {label}</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 94 |       <span className="text-[10px] text-[#C9CDD2]">· 전일 종가 기준</span>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 95 |     </div>
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 96 |   );
     | ^^^^ The component is created during render here
  97 |
  98 |   return (
  99 |     <>  react-hooks/static-components

/Users/bong/market-dashboard-v5/src/components/home/MorningBriefing.jsx
  46:7  warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  44 |     const stored = localStorage.getItem(DISMISS_KEY);
  45 |     if (stored === kstToday) {
> 46 |       setDismissed(true);
     |       ^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  47 |       setLoading(false);
  48 |       return;
  49 |     }  react-hooks/set-state-in-effect

/Users/bong/market-dashboard-v5/src/components/home/NotableMoversSection.jsx
  183:9  error  'hasClosedMarket' is assigned a value but never used. Allowed unused vars must match /^[A-Z_]/u  no-unused-vars

/Users/bong/market-dashboard-v5/src/components/home/TopNewsSection.jsx
   95:20  warning  Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

  93 |     const recent = allNews.filter(n => {
  94 |       if (!n.pubDate) return false;
> 95 |       try { return Date.now() - new Date(n.pubDate).getTime() < cutoff; }
     |                    ^^^^^^^^^^ Cannot call impure function
  96 |       catch { return false; }
  97 |     });
  98 |                                                                                                               react-hooks/purity
  113:22  warning  Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

  111 |       .sort((a, b) => {
  112 |         const TWO_HOURS = 2 * 3600000;
> 113 |         const aAge = Date.now() - new Date(a.pubDate).getTime();
      |                      ^^^^^^^^^^ Cannot call impure function
  114 |         const bAge = Date.now() - new Date(b.pubDate).getTime();
  115 |         const aFresh = aAge < TWO_HOURS;
  116 |         const bFresh = bAge < TWO_HOURS;                react-hooks/purity
  114:22  warning  Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

  112 |         const TWO_HOURS = 2 * 3600000;
  113 |         const aAge = Date.now() - new Date(a.pubDate).getTime();
> 114 |         const bAge = Date.now() - new Date(b.pubDate).getTime();
      |                      ^^^^^^^^^^ Cannot call impure function
  115 |         const aFresh = aAge < TWO_HOURS;
  116 |         const bFresh = bAge < TWO_HOURS;
  117 |         // 2시간 이내 기사는 최신순 우선 (시계열 정렬)  react-hooks/purity

/Users/bong/market-dashboard-v5/src/components/home/index.jsx
  45:20  warning  Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

  43 |     return allNews.filter(n => {
  44 |       if (!n.pubDate) return false;
> 45 |       try { return Date.now() - new Date(n.pubDate).getTime() < cutoff; } catch { return false; }
     |                    ^^^^^^^^^^ Cannot call impure function
  46 |     });
  47 |   }, [allNews]);
  48 |  react-hooks/purity

/Users/bong/market-dashboard-v5/src/hooks/useCoins.js
  49:3   warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  47 |   const wsConnectedRef  = useRef(false);
  48 |   const coinsRef      = useRef(coins);
> 49 |   coinsRef.current    = coins;
     |   ^^^^^^^^^^^^^^^^ Cannot update ref during render
  50 |
  51 |   // 마운트 시 snapshot 초기 로드
  52 |   useEffect(() => {  react-hooks/refs
  67:11  warning  Unused eslint-disable directive (no problems were reported from 'react-hooks/exhaustive-deps')

/Users/bong/market-dashboard-v5/src/hooks/useCompositeSignals.js
  166:11  warning  Unused eslint-disable directive (no problems were reported from 'react-hooks/exhaustive-deps')

/Users/bong/market-dashboard-v5/src/hooks/useDerivativeSignals.js
  25:3  warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  23 |   const krStocksRef = useRef(krStocks);
  24 |   const watchlistRef = useRef(watchlistSymbols);
> 25 |   usStocksRef.current = usStocks;
     |   ^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  26 |   krStocksRef.current = krStocks;
  27 |   watchlistRef.current = watchlistSymbols;
  28 |      react-hooks/refs
  26:3  warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  24 |   const watchlistRef = useRef(watchlistSymbols);
  25 |   usStocksRef.current = usStocks;
> 26 |   krStocksRef.current = krStocks;
     |   ^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  27 |   watchlistRef.current = watchlistSymbols;
  28 |
  29 |   useEffect(() => {                          react-hooks/refs
  27:3  warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  25 |   usStocksRef.current = usStocks;
  26 |   krStocksRef.current = krStocks;
> 27 |   watchlistRef.current = watchlistSymbols;
     |   ^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  28 |
  29 |   useEffect(() => {
  30 |     let pcrTimer, frTimer, ofTimer, socialTimer, vwapTimer, whaleTimer;  react-hooks/refs

/Users/bong/market-dashboard-v5/src/hooks/useIndices.js
  44:5  warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  42 |
  43 |   useEffect(() => {
> 44 |     refreshIndices();
     |     ^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  45 |     refreshExchangeRate();
  46 |     const indicesId = setInterval(() => { if (!document.hidden) refreshIndices(); }, POLLING.SLOW);
  47 |     const rateId    = setInterval(() => { if (!document.hidden) refreshExchangeRate(); }, POLLING.NORMAL);  react-hooks/set-state-in-effect

/Users/bong/market-dashboard-v5/src/hooks/useInvestorSignals.js
   12:3  error    'removeSignalByTypeAndSymbol' is defined but never used. Allowed unused vars must match /^[A-Z_]/u  no-unused-vars
  189:3  warning  Unused eslint-disable directive (no problems were reported from 'react-hooks/exhaustive-deps')

/Users/bong/market-dashboard-v5/src/hooks/useNewsSignals.js
  46:3  warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  44 |   // 이전 스캔에서 클러스터 시그널이 있던 심볼 추적
  45 |   const prevClusteredRef = useRef(new Set());
> 46 |   newsRef.current = allNews;
     |   ^^^^^^^^^^^^^^^ Cannot update ref during render
  47 |   itemsRef.current = allItems;
  48 |
  49 |   const scan = useCallback(() => {     react-hooks/refs
  47:3  warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  45 |   const prevClusteredRef = useRef(new Set());
  46 |   newsRef.current = allNews;
> 47 |   itemsRef.current = allItems;
     |   ^^^^^^^^^^^^^^^^ Cannot update ref during render
  48 |
  49 |   const scan = useCallback(() => {
  50 |     const news = newsRef.current;  react-hooks/refs

/Users/bong/market-dashboard-v5/src/hooks/usePrices.js
   57:3   warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  55 |   const krStocksRef = useRef(krStocks);
  56 |   const usStocksRef = useRef(usStocks);
> 57 |   krStocksRef.current = krStocks;
     |   ^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  58 |   usStocksRef.current = usStocks;
  59 |
  60 |   // 최신 watchlist 심볼 — 클로저 없이 참조 (App이 주입)  react-hooks/refs
   58:3   warning  Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  56 |   const usStocksRef = useRef(usStocks);
  57 |   krStocksRef.current = krStocks;
> 58 |   usStocksRef.current = usStocks;
     |   ^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  59 |
  60 |   // 최신 watchlist 심볼 — 클로저 없이 참조 (App이 주입)
  61 |   const krSymbolsRef = useRef([]);       react-hooks/refs
  190:11  warning  Unused eslint-disable directive (no problems were reported from 'react-hooks/exhaustive-deps')

/Users/bong/market-dashboard-v5/src/hooks/useSignals.js
   2:31  error    'useCallback' is defined but never used. Allowed unused vars must match /^[A-Z_]/u                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         no-unused-vars
  19:5   warning  Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  17 |     subscribe(handler);
  18 |     // 마운트 시점 최신 상태 반영
> 19 |     setSignals(getActiveSignals());
     |     ^^^^^^^^^^ Avoid calling setState() directly within an effect
  20 |     return () => unsubscribe(handler);
  21 |   }, []);
  22 |  react-hooks/set-state-in-effect

✖ 65 problems (24 errors, 41 warnings)
  0 errors and 5 warnings potentially fixable with the `--fix` option. 실패: 저장소 전반의 기존 lint 오류/경고 다수로 이번 수정 범위만의 신규 실패로 보긴 어려움

## 변경 방향
- 롤링 배너에 전용 viewport 래퍼 추가 및 ticker track 스타일 보강
- 시장온도 계산에 freshness/coverage 보정 및 시장별 trimmed fallback 도입
- 시그널 부족/지연 시 가격 흐름과 blended 표시
- 관심종목 모바일 칩에서 강제  제거
- 코인 로고 fallback 체인 강화 + 로고 상태 리셋 안정화
- 시그널보드 리스트는 심볼 기준 dedupe

--- a/src/components/SurgeBanner.jsx
+++ b/src/components/SurgeBanner.jsx
@@ -148,7 +148,7 @@ const SurgeBanner = memo(function SurgeBanner({ stocks = [], coins = [], indices
       </div>
 
       <div className="flex-1 min-w-0 overflow-hidden">
-        {/* ticker track */}
+        {/* 티커 이동 영역 */}
         <div className="ticker-track" style={{ '--dur': `${dur}s` }}>
           {items.map((item, i) => {
             // afterHours 표시: 정규장 외 시간에 시간외 데이터가 있을 때

--- a/src/components/SurgeBanner.jsx
+++ b/src/components/SurgeBanner.jsx
@@ -147,78 +147,80 @@ const SurgeBanner = memo(function SurgeBanner({ stocks = [], coins = [], indices
         </span>
       </div>
 
-      {/* ticker track */}
-      <div className="ticker-track" style={{ '--dur': `${dur}s` }}>
-        {items.map((item, i) => {
-          // afterHours 표시: 정규장 외 시간에 시간외 데이터가 있을 때
-          const showAfterHours = item.afterHoursPrice != null && item.afterHoursPct != null;
-          const displayPct     = showAfterHours ? item.afterHoursPct : item.pct;
-          const isUp           = displayPct > 0;
-          const isDown         = displayPct < 0;
+      <div className="flex-1 min-w-0 overflow-hidden">
+        {/* ticker track */}
+        <div className="ticker-track" style={{ '--dur': `${dur}s` }}>
+          {items.map((item, i) => {
+            // afterHours 표시: 정규장 외 시간에 시간외 데이터가 있을 때
+            const showAfterHours = item.afterHoursPrice != null && item.afterHoursPct != null;
+            const displayPct     = showAfterHours ? item.afterHoursPct : item.pct;
+            const isUp           = displayPct > 0;
+            const isDown         = displayPct < 0;
 
-          return (
-            <button
-              key={`${i}-${item.symbol}`}
-              type="button"
-              className="inline-flex items-center gap-1.5 px-3 h-full cursor-pointer transition-opacity hover:opacity-70 focus:outline-none bg-transparent border-0"
-              onClick={() => item._raw && onClick?.(item._raw)}
-            >
-              {/* 시장 컬러 도트 */}
-              <span
-                className="w-1.5 h-1.5 rounded-full flex-shrink-0"
-                style={{ background: MARKET_DOT[item.market] ?? C.dotIdleSolid }}
-              />
-
-              {/* 종목명 (국내는 이름, 해외/코인은 심볼) */}
-              <span
-                className="text-[11px] font-bold tracking-tight"
-                style={{ color: C.symbolText }}
+            return (
+              <button
+                key={`${i}-${item.symbol}`}
+                type="button"
+                className="inline-flex items-center gap-1.5 px-3 h-full cursor-pointer transition-opacity hover:opacity-70 focus:outline-none bg-transparent border-0 whitespace-nowrap"
+                onClick={() => item._raw && onClick?.(item._raw)}
               >
-                {item.market === 'kr' ? (item.name || item.symbol) : item.symbol}
-              </span>
-
-              {/* 지수 모드: 지수 값 */}
-              {item._value > 0 && (
+                {/* 시장 컬러 도트 */}
                 <span
-                  className="text-[11px] font-mono tabular-nums"
-                  style={{ color: C.valueText }}
-                >
-                  {item._value >= 1000
-                    ? item._value.toLocaleString('ko-KR', { maximumFractionDigits: 0 })
-                    : item._value.toFixed(2)}
-                </span>
-              )}
+                  className="w-1.5 h-1.5 rounded-full flex-shrink-0"
+                  style={{ background: MARKET_DOT[item.market] ?? C.dotIdleSolid }}
+                />
 
-              {/* 시간외 배지 */}
-              {showAfterHours && (
+                {/* 종목명 (국내는 이름, 해외/코인은 심볼) */}
                 <span
-                  className="text-[9px] font-semibold px-1 rounded"
-                  style={{
-                    background: C.afterHoursBg,
-                    color: C.afterHoursText,
-                    letterSpacing: '0.04em',
-                  }}
+                  className="text-[11px] font-bold tracking-tight"
+                  style={{ color: C.symbolText }}
                 >
-                  시간외
+                  {item.market === 'kr' ? (item.name || item.symbol) : item.symbol}
                 </span>
-              )}
 
-              {/* 등락률 */}
-              <span
-                className="text-[11px] font-semibold tabular-nums font-mono"
-                style={{ color: isUp ? C.pctUp : isDown ? C.pctDown : C.pctFlat }}
-              >
-                {isUp ? '+' : ''}{displayPct.toFixed(2)}%
-              </span>
+                {/* 지수 모드: 지수 값 */}
+                {item._value > 0 && (
+                  <span
+                    className="text-[11px] font-mono tabular-nums"
+                    style={{ color: C.valueText }}
+                  >
+                    {item._value >= 1000
+                      ? item._value.toLocaleString('ko-KR', { maximumFractionDigits: 0 })
+                      : item._value.toFixed(2)}
+                  </span>
+                )}
 
-              {/* 구분선 */}
-              <span
-                className="w-px h-3 flex-shrink-0 ml-1"
-                style={{ background: C.separator }}
-              />
-            </button>
-          );
-        })}
+                {/* 시간외 배지 */}
+                {showAfterHours && (
+                  <span
+                    className="text-[9px] font-semibold px-1 rounded"
+                    style={{
+                      background: C.afterHoursBg,
+                      color: C.afterHoursText,
+                      letterSpacing: '0.04em',
+                    }}
+                  >
+                    시간외
+                  </span>
+                )}
+
+                {/* 등락률 */}
+                <span
+                  className="text-[11px] font-semibold tabular-nums font-mono"
+                  style={{ color: isUp ? C.pctUp : isDown ? C.pctDown : C.pctFlat }}
+                >
+                  {isUp ? '+' : ''}{displayPct.toFixed(2)}%
+                </span>
+
+                {/* 구분선 */}
+                <span
+                  className="w-px h-3 flex-shrink-0 ml-1"
+                  style={{ background: C.separator }}
+                />
+              </button>
+            );
+          })}
+        </div>
       </div>
     </div>
   );

--- a/src/components/home/CommandCenterWidget.jsx
+++ b/src/components/home/CommandCenterWidget.jsx
@@ -2,7 +2,7 @@
 import { useMemo } from 'react';
 import { useSignals, useTopSignals } from '../../hooks/useSignals';
 import { useFearGreed } from '../../hooks/useFearGreed';
-import { calcTemperature, calcFallbackTemperature } from '../../utils/temperature';
+import { calcTemperature, calcFallbackTemperature, mergeTemperature } from '../../utils/temperature';
 import { extractName, getEasyLabel } from '../../utils/signalLabel';
 import { TYPE_META } from '../../engine/signalTypes';
 import MarketIndexSection from './MarketIndexSection';
@@ -28,8 +28,7 @@ function TemperatureBar({ indices, krwRate, allItems }) {
   const fallback = useMemo(() => calcFallbackTemperature(allItems), [allItems]);
   const { crypto, us, kr } = useFearGreed();
 
-  const isFallback = temp.count === 0;
-  const displayTemp = isFallback && fallback ? { ...temp, score: fallback.score, label: fallback.label } : temp;
+  const displayTemp = useMemo(() => mergeTemperature(temp, fallback), [temp, fallback]);
   const zone = ZONE[displayTemp.label] || ZONE['중립'];
   const gaugeWidth = Math.round(((displayTemp.score + 1) / 2) * 100);
 
@@ -50,6 +49,11 @@ function TemperatureBar({ indices, krwRate, allItems }) {
         <span className="text-[13px] font-bold flex-shrink-0" style={{ color: zone.text }}>
           {displayTemp.label} {gaugeWidth}%
         </span>
+        {displayTemp.source === 'blended' && (
+          <span className="text-[11px] font-medium flex-shrink-0 text-[#8B95A1]">
+            실시간 보정
+          </span>
+        )}
         {fgScore != null && (
           <span className="text-[12px] font-semibold flex-shrink-0 px-2.5 py-0.5 rounded-xl" style={{ background: 'rgba(255,149,0,0.06)', color: '#FF9500' }}>
             탐욕 {fgScore}
@@ -266,7 +270,8 @@ function WatchlistMini({ watchedItems, popularItems, toggle, onItemClick }) {
                   className="flex-shrink-0 flex items-center gap-1.5 px-2.5 py-1.5 rounded-full border border-[#F2F3F5] cursor-pointer hover:bg-[#F2F4F6] transition-colors"
                   onClick={() => onItemClick?.(item)}
                 >
-                  <span className="text-[11px] font-semibold text-[#191F28] whitespace-nowrap">{item.name?.slice(0, 6)}</span>
+                  <TickerLogo item={item} size={18} />
+                  <span className="text-[11px] font-semibold text-[#191F28] whitespace-nowrap max-w-[96px] truncate">{item.name || item.symbol}</span>
                   <span className="text-[10px] font-bold tabular-nums font-mono whitespace-nowrap" style={{ color }}>
                     {isUp ? '+' : ''}{(Number.isFinite(pct) ? pct : 0).toFixed(1)}%
                   </span>

--- a/src/components/home/SignalBoardWidget.jsx
+++ b/src/components/home/SignalBoardWidget.jsx
@@ -49,9 +49,17 @@ export default function SignalBoardWidget({ onItemClick }) {
   // 통합 리스트: 모든 시그널 (세력 포착 포함)
   const combinedList = useMemo(() => {
     const all = [...bullSignals, ...bearSignals, ...neutralSignals];
-    // 강도순 재정렬
-    all.sort((a, b) => (b.strength || 0) - (a.strength || 0));
-    return all;
+    all.sort((a, b) => (b.strength || 0) - (a.strength || 0) || (b.timestamp || 0) - (a.timestamp || 0));
+
+    const deduped = [];
+    const seen = new Set();
+    for (const signal of all) {
+      const dedupeKey = signal.symbol || `${signal.market || 'meta'}:${signal.type}:${signal.name || signal.label || ''}`;
+      if (seen.has(dedupeKey)) continue;
+      seen.add(dedupeKey);
+      deduped.push(signal);
+    }
+    return deduped;
   }, [bullSignals, bearSignals, neutralSignals]);
 
   const displayList = expanded ? combinedList : combinedList.slice(0, 5);

--- a/src/components/home/SignalBoardWidget.jsx
+++ b/src/components/home/SignalBoardWidget.jsx
@@ -50,16 +50,7 @@ export default function SignalBoardWidget({ onItemClick }) {
   const combinedList = useMemo(() => {
     const all = [...bullSignals, ...bearSignals, ...neutralSignals];
     all.sort((a, b) => (b.strength || 0) - (a.strength || 0) || (b.timestamp || 0) - (a.timestamp || 0));
-
-    const deduped = [];
-    const seen = new Set();
-    for (const signal of all) {
-      const dedupeKey = signal.symbol || `${signal.market || 'meta'}:${signal.type}:${signal.name || signal.label || ''}`;
-      if (seen.has(dedupeKey)) continue;
-      seen.add(dedupeKey);
-      deduped.push(signal);
-    }
-    return deduped;
+    return all;
   }, [bullSignals, bearSignals, neutralSignals]);
 
   const displayList = expanded ? combinedList : combinedList.slice(0, 5);

--- a/src/components/home/TickerLogo.jsx
+++ b/src/components/home/TickerLogo.jsx
@@ -1,5 +1,5 @@
 // 공통 종목 로고 컴포넌트 — getLogoUrls fallback 체인 + 이니셜 아바타
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 import { getLogoUrls, getAvatarBg } from './utils';
 
 /**
@@ -7,9 +7,11 @@ import { getLogoUrls, getAvatarBg } from './utils';
  * size: px 단위 (기본 24)
  */
 export default function TickerLogo({ item, size = 24 }) {
-  const logoUrls = getLogoUrls(item);
-  const [logoIdx, setLogoIdx] = useState(0);
+  const logoUrls = useMemo(() => getLogoUrls(item), [item]);
   const bg = getAvatarBg(item.symbol);
+  const logoKey = `${item.id || ''}:${item.symbol || ''}:${item.image || ''}:${item._market || item.market || ''}`;
+  const [logoState, setLogoState] = useState(() => ({ key: logoKey, idx: 0 }));
+  const logoIdx = logoState.key === logoKey ? logoState.idx : 0;
 
   const sizeStyle = { width: size, height: size, minWidth: size, minHeight: size };
   const fontSize = Math.max(8, Math.round(size * 0.375));
@@ -19,7 +21,12 @@ export default function TickerLogo({ item, size = 24 }) {
       <img
         src={logoUrls[logoIdx]}
         alt={item.symbol}
-        onError={() => setLogoIdx(i => i + 1)}
+        onError={() => {
+          setLogoState(prev => ({
+            key: logoKey,
+            idx: (prev.key === logoKey ? prev.idx : 0) + 1,
+          }));
+        }}
         className="rounded-full object-contain bg-white border border-[#F2F4F6] p-0.5 flex-shrink-0"
         style={sizeStyle}
       />

--- a/src/components/home/utils.js
+++ b/src/components/home/utils.js
@@ -116,7 +116,10 @@ export function getLogoUrls(item) {
     // 코인: 원본 이미지 → CoinPaprika → CoinCap → cryptocurrency-icons
     const urls = [];
     if (item.image) urls.push(item.image);
-    const paprikaId = item.id?.includes('-') ? item.id : COIN_PAPRIKA_IDS[sym.toUpperCase()];
+    const normalizedSymbol = sym.toLowerCase();
+    const paprikaId = item.id?.toLowerCase()?.startsWith(`${normalizedSymbol}-`)
+      ? item.id
+      : COIN_PAPRIKA_IDS[sym.toUpperCase()];
     if (paprikaId) urls.push(`https://static.coinpaprika.com/coin/${paprikaId}/logo.png`);
     urls.push(`https://assets.coincap.io/assets/icons/${sym.toLowerCase()}@2x.png`);
     urls.push(`https://cdn.jsdelivr.net/gh/spothq/cryptocurrency-icons@master/128/color/${sym.toLowerCase()}.png`);

--- a/src/components/home/utils.js
+++ b/src/components/home/utils.js
@@ -83,6 +83,29 @@ export function getAvatarBg(symbol) {
   return PALETTE[Math.abs((symbol || '').split('').reduce((h, c) => c.charCodeAt(0) + ((h << 5) - h), 0)) % PALETTE.length] || '#8B95A1';
 }
 
+const COIN_PAPRIKA_IDS = {
+  BTC: 'btc-bitcoin',
+  ETH: 'eth-ethereum',
+  XRP: 'xrp-xrp',
+  SOL: 'sol-solana',
+  DOGE: 'doge-dogecoin',
+  ADA: 'ada-cardano',
+  BNB: 'bnb-binance-coin',
+  AVAX: 'avax-avalanche',
+  DOT: 'dot-polkadot',
+  LINK: 'link-chainlink',
+  TON: 'ton-toncoin',
+  TRX: 'trx-tron',
+  BCH: 'bch-bitcoin-cash',
+  LTC: 'ltc-litecoin',
+  SHIB: 'shib-shiba-inu',
+  UNI: 'uni-uniswap',
+  ATOM: 'atom-cosmos',
+  ETC: 'etc-ethereum-classic',
+  XLM: 'xlm-stellar',
+  HBAR: 'hbar-hedera-hashgraph',
+};
+
 // ─── 종목/코인 로고 URL fallback 체인 ─────────────────────────
 // 반환: [url1, url2, ...] — 순서대로 시도, 모두 실패 시 이니셜 아바타
 export function getLogoUrls(item) {
@@ -90,10 +113,13 @@ export function getLogoUrls(item) {
   const market = item._market || (item.market === 'coin' ? 'COIN' : item.market === 'kr' ? 'KR' : item.market === 'us' ? 'US' : '');
 
   if (market === 'COIN' || item.id) {
-    // 코인: CoinGecko → CoinCap → cryptocurrency-icons
+    // 코인: 원본 이미지 → CoinPaprika → CoinCap → cryptocurrency-icons
     const urls = [];
     if (item.image) urls.push(item.image);
+    const paprikaId = item.id?.includes('-') ? item.id : COIN_PAPRIKA_IDS[sym.toUpperCase()];
+    if (paprikaId) urls.push(`https://static.coinpaprika.com/coin/${paprikaId}/logo.png`);
     urls.push(`https://assets.coincap.io/assets/icons/${sym.toLowerCase()}@2x.png`);
+    urls.push(`https://cdn.jsdelivr.net/gh/spothq/cryptocurrency-icons@master/128/color/${sym.toLowerCase()}.png`);
     urls.push(`https://raw.githubusercontent.com/spothq/cryptocurrency-icons/master/128/color/${sym.toLowerCase()}.png`);
     return urls;
   }

--- a/src/components/home/widgets/MarketSentimentWidget.jsx
+++ b/src/components/home/widgets/MarketSentimentWidget.jsx
@@ -133,7 +133,9 @@ export default function MarketSentimentWidget({ allItems = [] }) {
 
       {/* 한 줄 해석 */}
       <p className="text-[13px] font-medium text-[#333D4B] mb-3">
-        "{displayTemp.source !== 'signals' ? '시그널과 가격 흐름을 함께 반영한 온도에요' : message}"
+        "{displayTemp.source === 'blended' || displayTemp.source === 'fallback'
+          ? '시그널과 가격 흐름을 함께 반영한 온도에요'
+          : message}"
       </p>
 
       {/* 게이지 바 */}

--- a/src/components/home/widgets/MarketSentimentWidget.jsx
+++ b/src/components/home/widgets/MarketSentimentWidget.jsx
@@ -4,7 +4,7 @@ import { useState, useMemo } from 'react';
 import { useSignals } from '../../../hooks/useSignals';
 import { useFearGreed, getFgColor } from '../../../hooks/useFearGreed';
 import { TYPE_META } from '../../../engine/signalTypes';
-import { calcTemperature, calcFallbackTemperature } from '../../../utils/temperature';
+import { calcTemperature, calcFallbackTemperature, mergeTemperature } from '../../../utils/temperature';
 
 // ── 게이지 존 스타일 (5단계) ──
 const ZONE = {
@@ -75,9 +75,7 @@ export default function MarketSentimentWidget({ allItems = [] }) {
   const fallback = useMemo(() => calcFallbackTemperature(allItems), [allItems]);
   const { crypto, us, kr } = useFearGreed();
 
-  // 시그널 0건일 때 가격 기반 fallback 사용 여부
-  const isFallback = temp.count === 0;
-  const displayTemp = isFallback && fallback ? { ...temp, score: fallback.score, label: fallback.label } : temp;
+  const displayTemp = useMemo(() => mergeTemperature(temp, fallback), [temp, fallback]);
   const zone = ZONE[displayTemp.label] || ZONE['중립'];
   const message = MESSAGE_MAP[displayTemp.label] || MESSAGE_MAP['중립'];
   const gaugeWidth = Math.round(((displayTemp.score + 1) / 2) * 100);
@@ -96,7 +94,7 @@ export default function MarketSentimentWidget({ allItems = [] }) {
   }, [signals]);
 
   // 시그널 0건 + fallback도 없으면 스켈레톤
-  if (isFallback && !fallback) {
+  if (temp.count === 0 && !fallback) {
     return (
       <div
         data-testid="market-sentiment"
@@ -135,7 +133,7 @@ export default function MarketSentimentWidget({ allItems = [] }) {
 
       {/* 한 줄 해석 */}
       <p className="text-[13px] font-medium text-[#333D4B] mb-3">
-        "{isFallback ? '시그널 수집 중... 가격 기준 임시 온도에요' : message}"
+        "{displayTemp.source !== 'signals' ? '시그널과 가격 흐름을 함께 반영한 온도에요' : message}"
       </p>
 
       {/* 게이지 바 */}

--- a/src/index.css
+++ b/src/index.css
@@ -151,6 +151,9 @@
   .ticker-wrap { overflow: hidden; }
   .ticker-track {
     display: inline-flex;
+    width: max-content;
+    white-space: nowrap;
+    will-change: transform;
     animation: ticker var(--dur, 40s) linear infinite;
   }
   .ticker-track:hover { animation-play-state: paused; }

--- a/src/utils/temperature.js
+++ b/src/utils/temperature.js
@@ -120,11 +120,15 @@ export function calcFallbackTemperature(allItems) {
     recentDirectionalCount: marketScores.length * 2,
   });
 
-  return { score, label, avgPct, marketScores };
+  return { score, label, avgPct, marketScores, partial: marketScores.length < 2 };
 }
 
 export function mergeTemperature(signalTemp, fallbackTemp) {
   if (!fallbackTemp) return { ...signalTemp, source: 'signals' };
+  if (fallbackTemp.partial) {
+    if (!signalTemp?.count) return { score: 0, label: '중립', count: 0, source: 'pending' };
+    return { ...signalTemp, source: 'signals' };
+  }
   if (!signalTemp?.count) return { ...fallbackTemp, count: 0, source: 'fallback' };
 
   const directionalCount = signalTemp.directionalCount ?? 0;

--- a/src/utils/temperature.js
+++ b/src/utils/temperature.js
@@ -1,46 +1,148 @@
 // 시장 온도 계산 유틸리티 — MarketSentimentWidget에서 추출
 import { getPct } from '../components/home/utils';
 
+function labelFromScore(score, { directionalCount = 0, recentDirectionalCount = 0 } = {}) {
+  const hasModerateCoverage = directionalCount >= 4;
+  const hasStrongCoverage = directionalCount >= 8 && recentDirectionalCount >= 3;
+
+  if (score <= -0.6) return hasStrongCoverage ? '강한 경계' : '약세 우위';
+  if (score <= -0.22) return hasModerateCoverage ? '약세 우위' : '중립';
+  if (score < 0.22) return '중립';
+  if (score < 0.6) return hasModerateCoverage ? '강세 징후' : '중립';
+  return hasStrongCoverage ? '강한 강세' : '강세 징후';
+}
+
 // ── 시그널 기반 온도 계산 ──
 export function calcTemperature(signals) {
-  if (!signals.length) return { score: 0, label: '중립', count: 0, bullCount: 0, bearCount: 0, neutralCount: 0 };
-  let bullWeight = 0, bearWeight = 0, neutralCount = 0;
-  for (const sig of signals) {
-    const w = sig.strength || 1;
-    if (sig.direction === 'bullish') bullWeight += w;
-    else if (sig.direction === 'bearish') bearWeight += w;
-    else neutralCount++;
+  if (!signals.length) {
+    return {
+      score: 0,
+      label: '중립',
+      count: 0,
+      bullCount: 0,
+      bearCount: 0,
+      neutralCount: 0,
+      directionalCount: 0,
+      recentDirectionalCount: 0,
+      freshness: 0,
+      latestSignalAt: null,
+    };
   }
+
+  const now = Date.now();
+  const recentWindowMs = 3 * 60 * 1000;
+  let bullWeight = 0;
+  let bearWeight = 0;
+  let neutralCount = 0;
+  let bullCount = 0;
+  let bearCount = 0;
+  let directionalCount = 0;
+  let recentDirectionalCount = 0;
+  let freshnessSum = 0;
+  let latestSignalAt = 0;
+
+  for (const sig of signals) {
+    const createdAt = sig.timestamp ?? now;
+    const ttl = Math.max(1, (sig.expiresAt ?? now) - createdAt);
+    const age = Math.max(0, now - createdAt);
+    const freshness = Math.max(0.35, 1 - (age / ttl) * 0.65);
+    const weight = (sig.strength || 1) * freshness;
+
+    latestSignalAt = Math.max(latestSignalAt, createdAt);
+    freshnessSum += freshness;
+
+    if (sig.direction === 'bullish') {
+      bullWeight += weight;
+      bullCount++;
+      directionalCount++;
+      if (age <= recentWindowMs) recentDirectionalCount++;
+    } else if (sig.direction === 'bearish') {
+      bearWeight += weight;
+      bearCount++;
+      directionalCount++;
+      if (age <= recentWindowMs) recentDirectionalCount++;
+    } else {
+      neutralCount++;
+    }
+  }
+
   const total = bullWeight + bearWeight;
-  const score = total === 0 ? 0 : (bullWeight - bearWeight) / total;
-  // 시그널 5개 미만이면 극단 라벨 방지 — 최대 "중립"까지만 허용
-  // (부팅 시드 3개만으로 "강한 강세" 표시 방지)
-  const hasEnoughSignals = signals.length >= 5;
-  let label;
-  if (score <= -0.5) label = hasEnoughSignals ? '강한 경계' : '중립';
-  else if (score <= -0.15) label = hasEnoughSignals ? '약세 우위' : '중립';
-  else if (score < 0.15) label = '중립';
-  else if (score < 0.5) label = hasEnoughSignals ? '강세 징후' : '중립';
-  else label = hasEnoughSignals ? '강한 강세' : '중립';
-  return { score, label, count: signals.length,
-    bullCount: signals.filter(s => s.direction === 'bullish').length,
-    bearCount: signals.filter(s => s.direction === 'bearish').length,
-    neutralCount };
+  const rawScore = total === 0 ? 0 : (bullWeight - bearWeight) / total;
+  const coverage = Math.min(1, directionalCount / 8);
+  const recency = Math.min(1, recentDirectionalCount / 4);
+  const confidence = directionalCount === 0 ? 0 : Math.max(0.3, (coverage * 0.65) + (recency * 0.35));
+  const score = rawScore * confidence;
+  const label = labelFromScore(score, { directionalCount, recentDirectionalCount });
+
+  return {
+    score,
+    rawScore,
+    label,
+    count: signals.length,
+    bullCount,
+    bearCount,
+    neutralCount,
+    directionalCount,
+    recentDirectionalCount,
+    freshness: signals.length ? freshnessSum / signals.length : 0,
+    latestSignalAt: latestSignalAt || null,
+  };
 }
 
 // ── 가격 기반 fallback 온도 계산 ──
 export function calcFallbackTemperature(allItems) {
   if (!allItems?.length) return null;
-  const pcts = allItems.map(i => getPct(i)).filter(p => !isNaN(p));
-  if (!pcts.length) return null;
-  const avg = pcts.reduce((a, b) => a + b, 0) / pcts.length;
-  // 평균 등락률을 -1 ~ +1 스코어로 변환 (+-5% 기준 클램핑)
-  const score = Math.max(-1, Math.min(1, avg / 5));
-  let label;
-  if (score <= -0.5) label = '강한 경계';
-  else if (score <= -0.15) label = '약세 우위';
-  else if (score < 0.15) label = '중립';
-  else if (score < 0.5) label = '강세 징후';
-  else label = '강한 강세';
-  return { score, label, avgPct: avg };
+  const groups = { KR: [], US: [], COIN: [] };
+
+  for (const item of allItems) {
+    const pct = getPct(item);
+    if (!Number.isFinite(pct)) continue;
+    const market = item._market || (item.id || item.market === 'coin' ? 'COIN' : item.market === 'kr' ? 'KR' : item.market === 'us' ? 'US' : null);
+    if (market && groups[market]) groups[market].push(pct);
+  }
+
+  const activeMarkets = Object.entries(groups).filter(([, values]) => values.length > 0);
+  if (!activeMarkets.length) return null;
+
+  const marketScores = activeMarkets.map(([market, values]) => {
+    const sorted = [...values].sort((a, b) => a - b);
+    const trim = sorted.length >= 8 ? Math.floor(sorted.length * 0.15) : 0;
+    const trimmed = trim > 0 ? sorted.slice(trim, sorted.length - trim) : sorted;
+    const avg = trimmed.reduce((sum, value) => sum + value, 0) / trimmed.length;
+    const score = Math.max(-1, Math.min(1, avg / 3.5));
+    return { market, avgPct: avg, score, count: values.length };
+  });
+
+  const score = marketScores.reduce((sum, market) => sum + market.score, 0) / marketScores.length;
+  const avgPct = marketScores.reduce((sum, market) => sum + market.avgPct, 0) / marketScores.length;
+  const label = labelFromScore(score, {
+    directionalCount: marketScores.length * 3,
+    recentDirectionalCount: marketScores.length * 2,
+  });
+
+  return { score, label, avgPct, marketScores };
+}
+
+export function mergeTemperature(signalTemp, fallbackTemp) {
+  if (!fallbackTemp) return { ...signalTemp, source: 'signals' };
+  if (!signalTemp?.count) return { ...fallbackTemp, count: 0, source: 'fallback' };
+
+  const directionalCount = signalTemp.directionalCount ?? 0;
+  const recentDirectionalCount = signalTemp.recentDirectionalCount ?? 0;
+  const shouldBlend = directionalCount < 6 || recentDirectionalCount < 3;
+  if (!shouldBlend) return { ...signalTemp, source: 'signals' };
+
+  const signalWeight = directionalCount < 4 ? 0.4 : 0.6;
+  const fallbackWeight = 1 - signalWeight;
+  const score = (signalTemp.score * signalWeight) + (fallbackTemp.score * fallbackWeight);
+  const label = labelFromScore(score, { directionalCount, recentDirectionalCount });
+
+  return {
+    ...signalTemp,
+    score,
+    label,
+    fallbackScore: fallbackTemp.score,
+    fallbackLabel: fallbackTemp.label,
+    source: 'blended',
+  };
 }

--- a/src/utils/temperature.js
+++ b/src/utils/temperature.js
@@ -1,5 +1,4 @@
 // 시장 온도 계산 유틸리티 — MarketSentimentWidget에서 추출
-import { getPct } from '../components/home/utils';
 
 function labelFromScore(score, { directionalCount = 0, recentDirectionalCount = 0 } = {}) {
   const hasModerateCoverage = directionalCount >= 4;
@@ -95,7 +94,9 @@ export function calcFallbackTemperature(allItems) {
   const groups = { KR: [], US: [], COIN: [] };
 
   for (const item of allItems) {
-    const pct = getPct(item);
+    const isCoin = !!(item.id || item._market === 'COIN' || item.market === 'coin');
+    const rawPct = isCoin ? item.change24h : item.changePct;
+    const pct = Number.isFinite(rawPct) ? rawPct : NaN;
     if (!Number.isFinite(pct)) continue;
     const market = item._market || (item.id || item.market === 'coin' ? 'COIN' : item.market === 'kr' ? 'KR' : item.market === 'us' ? 'US' : null);
     if (market && groups[market]) groups[market].push(pct);

